include/cooking/cookingQuerys/ratingSystem.php

```diff
@@ -0,0 +1,136 @@
+    dbQuery("INSERT INTO recipe_identifier_linked(recipeIngredientName, recipeId)
+        VALUES (:recipeIngredientName, :recipeId)", array('recipeIngredientName'=>$tagName, 'recipeId'=>$recipeId));
+            //header('Location:cookingWebsite/yourCookbook/yourRecipes.php');
+}
+function deleteTagFromExistingRecipes($tagName, $postId){
+    dbQuery("DELETE FROM recipe_identifier_linked WHERE recipeIngredientName=:recipeIngredientName and recipeId=:recipeId",
+        array('recipeIngredientName'=>$tagName, 'recipeId'=>$postId));
+        //header('Location:cookingWebsite/yourCookbook/yourRecipes.php');
+}
+function addNewCookingTagQuery ($tagName){
+        dbQuery("INSERT INTO recipeTagIds (recipeIngredientName)
+            VALUES (:recipeIngredientName)", array('recipeIngredientName'=>$tagName));
+            header('Location:/cookingWebsite/yourCookbook/addTags.php');
+}
+function deleteNewCookingTagQuery ($tagName){
+        dbQuery("DELETE FROM recipeTagIds WHERE recipeIngredientName=:recipeIngredientName", array('recipeIngredientName'=>$tagName));
+        dbQuery("DELETE FROM recipe_identifier_linked WHERE recipeIngredientName= :recipeIngredientName", array('recipeIngredientName'=>$tagName));
+        return header('Location:/cookingWebsite/yourCookbook/addTags.php');
+}
+
+// ********ALL FUNCTIONS FOR RATINGS BELOW*********
+
+function insertRatingIntoTable($loginId, $recipeId, $rating){
+    dbQuery("INSERT INTO ratings(loginId, recipeId, rating)
+            VALUES(:loginId, :recipeId, :rating)",
+            array('loginId'=>$loginId, 'recipeId'=>$recipeId, 'rating'=>$rating));
+}
+function algorithm(){
+    $total = 0;
+    $loginQuery = dbQuery("SELECT * FROM cookingLogin")->fetchALL();
+    $loginArray = array();
+        //$returnRecipeTags = "";
+        foreach ($loginQuery as $logins){
+            if(!@$loginArray[$logins['loginId']]){
+                $loginArray[$logins['loginId']] = true;
+            }
+                $loginId = $logins['loginId'];
+
+
+    $recipeQuery = dbQuery("SELECT * FROM recipes")->fetchALL();
+    $recipeArray = array();
+        //$returnRecipeTags = "";
+        foreach ($recipeQuery as $recipes){
+            if(!@$recipeArray[$recipes['recipeId']]){
+                $recipeArray[$recipes['recipeId']] = true;
+            }
+                $recipeId = $recipes['recipeId'];
+                $tagQuery = dbQuery("SELECT * FROM recipe_identifier_linked
+                            WHERE recipeId= :recipeId", array('recipeId'=>$recipeId))->fetchALL();
+                $tagArray = array();
+                foreach ($tagQuery as $tags){
+
+                    if(!@$tagArray[$tags['recipeIngredientName']]){
+                        $tagArray[$tags['recipeIngredientName']] = true;
+
+                    }
+                    $recipeIngredientName = $tags['recipeIngredientName'];
+                    $recipeTagQuery = dbQuery("SELECT * FROM recipe_identifier_linked
```

Comment:
In addition to helping performance, it'd be good to move these into their own section of the code because they're sort of cluttering up the logic of the algorithm so it's a bit harder to tell what you're trying to do.

Related to that, I think this code could benefit from comments. Algorithms can be hard to follow (even by you if you were to come back and read this a few months from now) so it can help to leave little comments explaining what you're doing. Things like:
```
//Get a list of a the recipes, and the ingredients for each recipe.

//Get a list of how the user rated each recipe.

//Loop through the user ratings and count each rating as a "score" for each ingredient in that recipe. We'll average the scores for different ingredients

//Now that we know how much a user likes each ingredient, we can use that to predict how the user will rate the other recipes
```

As code gets more complicated, you'll want to comment it like that because it's harder and harder to figure out on its own. This will also be a big help if you're ever working on code with other people so they can understand what you're doing.